pkg/deploy/controller/deployment/controller.go

```diff
@@ -277,40 +258,39 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 	return pod, nil
 }
 
-func (c *DeploymentController) cancelDeployerPods(deployment *kapi.ReplicationController) error {
+func (c *DeploymentController) cancelDeployerPods(deployment *kapi.ReplicationController, config *deployapi.DeploymentConfig) error {
 	deployerPods, err := c.podClient.getDeployerPodsFor(deployment.Namespace, deployment.Name)
 	if err != nil {
 		return fmt.Errorf("couldn't fetch deployer pods for %s while trying to cancel deployment: %v", deployutil.LabelForDeployment(deployment), err)
 	}
 	glog.V(4).Infof("Cancelling %d deployer pods for deployment %s", len(deployerPods), deployutil.LabelForDeployment(deployment))
 	zeroDelay := int64(1)
-	anyCancelled := false
+	cleanedAll := len(deployerPods) > 0
 	for _, deployerPod := range deployerPods {
 		// Set the ActiveDeadlineSeconds on the pod so it's terminated very soon.
 		if deployerPod.Spec.ActiveDeadlineSeconds == nil || *deployerPod.Spec.ActiveDeadlineSeconds != zeroDelay {
 			deployerPod.Spec.ActiveDeadlineSeconds = &zeroDelay
 			if _, err := c.podClient.updatePod(deployerPod.Namespace, &deployerPod); err != nil {
-				if config, decodeErr := c.decodeConfig(deployment); decodeErr == nil {
-					c.recorder.Eventf(config, kapi.EventTypeWarning, "FailedCancellation", "Error cancelling deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
-				} else {
-					c.recorder.Eventf(deployment, kapi.EventTypeWarning, "FailedCancellation", "Error cancelling deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
-				}
-				return fmt.Errorf("couldn't cancel deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+				cleanedAll := false
+				utilruntime.HandleError(fmt.Errorf("couldn't cancel deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err))
 			}
-			anyCancelled = true
 			glog.V(4).Infof("Cancelled deployer pod %s for deployment %s", deployerPod.Name, deployutil.LabelForDeployment(deployment))
 		}
 	}
-	if anyCancelled {
-		if config, decodeErr := c.decodeConfig(deployment); decodeErr == nil && len(deployerPods) > 0 {
-			c.recorder.Eventf(config, kapi.EventTypeNormal, "Cancelled", "Cancelled deployer pods for deployment %s", deployutil.LabelForDeployment(deployment))
-		} else if len(deployerPods) > 0 {
-			c.recorder.Eventf(deployment, kapi.EventTypeNormal, "Cancelled", "Cancelled deployer pods")
-		}
+	if cleanedAll {
+		c.emitDeploymentEvent(deployment, config, kapi.EventTypeNormal, "Cancelled", "Cancelled all deployer pods")
 	}
 	return nil
 }
 
+func (c *DeploymentController) emitDeploymentEvent(deployment *kapi.ReplicationController, config *deployapi.DeploymentConfig, eventType, title, message string) {
+	if config != nil {
+		c.recorder.Eventf(config, eventType, title, fmt.Sprintf("%s: %s", deployutil.LabelForDeployment(deployment), message))
```

Comment:
Since events already have namespaces, we should not use LabelForDeployment, we should just print the name of the deployment.